spec/metrics/upstream_spec.lua

```diff
@@ -36,8 +37,8 @@ describe('upstream metrics', function()
     end)
 
     it('adds the latency to the histogram', function()
-      upstream_metrics.report(200, 0.1)
-      assert.stub(test_histogram.observe).was_called_with(test_histogram, 0.1)
+      upstream_metrics.report(200, 0.1, service_metric_name)
```

Comment:
We should probably test with and without service. To make sure that both test cases work correctly.

Also, we should do the same for the status codes counter.